### PR TITLE
feat: use vuetify components in login views

### LINKED ISSUE #112: [HackathonTheme] - Setup User module views
Redesign the views of the User module to use the Material design look. 
- Include ExternalProvider logins

### PR DESCRIPTION
**Pull request by M-A-Boucher**
fix #112

### CODE REVIEW

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/ChangePassword.cshtml [hunk 1]
```diff
@@ -1,33 +1,36 @@
...
+          <v-text-field id="PasswordConfirmation" type="password" name="PasswordConfirmation" label="@T["New password confirmation"]" value="@passwordConfirmation?.AttemptedValue" error-messages="@(passwordConfirmation?.Errors.FirstOrDefault()?.ErrorMessage)"></v-text-field>
+        </v-col>
+      </v-row>
     </form>
-  </section>
-</div>
\ No newline at end of file
+  </v-card-text>
+  <v-card-actions>
+    <v-btn type="submit" form="form" x-large block color="success">@T["Change password"]</v-btn>
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/ChangePassword.cshtml:34]: What does the `form="form"` do ? I don't think it's required

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml [hunk 2]
```diff
@@ -22,73 +22,71 @@
...
+    </v-row>
+  </v-card-text>
+  @if (!disableLocalLogin)
+  {
+    <v-card-actions>
+      @await RenderSectionAsync("AfterLogin", required: false)
+      <v-btn type="submit" form="form" x-large block color="success">@T["Log in"]</v-btn>
+    </v-card-actions>
+  }
+  @if (loginProviders.Count == 0)
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml:64]: I think you forgot to revert this ;)

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml [hunk 3]
```diff
@@ -22,73 +22,71 @@
...
+    </v-card-actions>
+  }
+  @if (userCanRegister)
+  {
+    <div class="d-flex align-center">
+      <v-divider></v-divider><span class="px-2">@T["or"]</span><v-divider></v-divider>
+    </div>
+    <v-card-actions>
+      <form id="RegistrationForm" asp-controller="Registration" asp-action="Register" asp-route-returnurl="@ViewData["ReturnUrl"]" method="post"></form>
+      <v-btn type="submit" form="RegistrationForm" x-large block color="grey lighten-2">@T["Create an account"]</v-btn>
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml:89]: do you need the form="" here as well ?

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml [hunk 4]
```diff
@@ -6,61 +6,64 @@
...
+          </v-text-field>
+        </form>
+      </v-col>
+    </v-row>
+  </v-card-text>
+  <v-card-actions>
+    @await RenderSectionAsync("AfterRegister", required: false)
+    <v-btn type="submit" form="form" x-large block color="success"><span>@T["Create account"]</span></button>
+  </v-card-actions>
+  @if (!disableLocalLogin)
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml:56]: I wonder if we should display this if locallogin is enabled or externalprovider > 0

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml [hunk 5]
```diff
@@ -6,61 +6,64 @@
...
-        <div class="mt-4">
-          <a asp-route-area="OrchardCore.Users" asp-controller="Account" asp-action="Login">@T["Already have an account? Login"]</a>
-        </div>
-      }
-    </section>
-</div>
\ No newline at end of file
+    @* <span>@T["Already have an account?"]</span> *@
+    <v-card-actions>
+      <form id="LoginForm" asp-route-area="OrchardCore.Users" asp-controller="Account" asp-action="Login"></form>
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml:64]: Why is this a form ? this should be a simple link I think.

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/ResetPassword/ForgotPassword.cshtml [hunk 6]
```diff
@@ -1,25 +1,21 @@
...
+    <form id="form" asp-controller="ResetPassword" asp-action="ForgotPassword" method="post">
+      <v-text-field id="UserIdentifier" name="UserIdentifier"
+      label="@T["Enter your email or username to reset password."]" value="@userIdentifier?.AttemptedValue" error-messages="@(userIdentifier?.Errors.FirstOrDefault()?.ErrorMessage)"></v-text-field>
     </form>
-  </section>
-</div>
+  </v-card-text>
+  <v-card-actions>
+    @await RenderSectionAsync("AfterForgotPassword", required: false)
+    <v-btn type="submit" form="form" x-large block color="success">@T["Send recovery email"]</v-btn>
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/OrchardCore.Users/ResetPassword/ForgotPassword.cshtml:19]: remove form="form" (I think)

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/ResetPassword/ResetPassword.cshtml [hunk 7]
```diff
@@ -2,46 +2,49 @@
...
+          </ul>
+        </v-alert>
+      }
+      <v-row>
+        <v-col cols="12">
+          <v-text-field id="Email" type="email" name="Email" label="@T["Email"]" value="@email?.AttemptedValue" error-messages="@(email?.Errors.FirstOrDefault()?.ErrorMessage)"></v-text-field>
+          <v-row>
+            <v-col cols="auto">
+              <v-text-field id="NewPassword" type="password" name="NewPassword" label="@T["New password"]" value="@newPassword?.AttemptedValue" error-messages="@(newPassword?.Errors.FirstOrDefault()?.ErrorMessage)"></v-text-field>
+              <v-text-field id="PasswordConfirmation" type="password" name="PasswordConfirmation" label="@T["New password confirmation"]" value="@passwordConfirmation?.AttemptedValue" error-messages="@(passwordConfirmation?.Errors.FirstOrDefault()?.ErrorMessage)"></v-text-field>
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/OrchardCore.Users/ResetPassword/ResetPassword.cshtml:32]: I may be wrong, but usually you don't want to pass the value back to password fields. Maybe asp.net does this by default ? not sure. In any case, I think we should remove all values for passwords as a security measure

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/ResetPassword/ResetPassword.cshtml [hunk 8]
```diff
@@ -2,46 +2,49 @@
...
-          <button type="submit" class="btn btn-lg btn-primary is-rounded">@T["Reset password"]</button>
-      </div>
     </form>
-  </section>
-</div>
\ No newline at end of file
+  </v-card-text>
+  <v-card-actions>
+    @await RenderSectionAsync("AfterResetPassword", required: false)
+    <v-btn type="submit" form="form" x-large block color="success">@T["Reset password"]</v-btn>
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/OrchardCore.Users/ResetPassword/ResetPassword.cshtml:48]: form="form"

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml [hunk 9]
```diff
@@ -22,73 +22,71 @@
...
-  </v-card-text>
+    </v-card-actions>
+  }
+  @if (userCanRegister)
+  {
+    <div class="d-flex align-center">
+      <v-divider></v-divider><span class="px-2">@T["or"]</span><v-divider></v-divider>
+    </div>
+    <v-card-actions>
+      <form id="RegistrationForm" asp-controller="Registration" asp-action="Register" asp-route-returnurl="@ViewData["ReturnUrl"]" method="post"></form>
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml:88]: This should be a link.

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml [hunk 10]
```diff
@@ -6,61 +6,64 @@
...
+                  <li>@error.ErrorMessage</li>
+                }
+              </ul>
+            </v-alert>
+          }
+          <v-text-field id="UserName" name="UserName" label="@T["Username"]" value="@userName?.AttemptedValue"
+          error-messages="@(userName?.Errors.FirstOrDefault()?.ErrorMessage)"></v-text-field>
+          <v-text-field id="Email" type="email" name="Email" label="@T["Email"]" value="@email?.AttemptedValue"
+          error-messages="@(email?.Errors.FirstOrDefault()?.ErrorMessage)"></v-text-field>
+          <v-text-field id="Password" type="password" name="Password" label="@T["Password"]"
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml:41]: Remove value from password fields

**Review by jptissot — CHANGES_REQUESTED** [review]

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/ChangePassword.cshtml (hunk 1 above)
Comment (M-A-Boucher) [reply src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/ChangePassword.cshtml:34]: I put the submit button inside v-card-actions outside of the form's influence. That's the only reason why it's required. I could wrap the entire v-card with form instead?

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml (hunk 2 above)
Comment (M-A-Boucher) [reply src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml:64]: Yep. :)

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml (hunk 3 above)
Comment (M-A-Boucher) [reply src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml:89]: This one can easily be wrapped.

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml (hunk 4 above)
Comment (M-A-Boucher) [reply src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml:56]: Probably shouldn't display. I'll put the check.

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml (hunk 5 above)
Comment (M-A-Boucher) [reply src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml:64]: It was a bit of a straight port. If you think a link to the login screen will be enough, then I will make it a link.

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml (hunk 9 above)
Comment (M-A-Boucher) [reply src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml:88]: I think the ReturnlUrl functionality is needed to return to the login screen after registering. A link wouldn't give this without some extra param tacked on...

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/ChangePassword.cshtml (hunk 1 above)
Comment (jptissot) [reply src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/ChangePassword.cshtml:34]: Ah ok, cool. Just wondering.

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml [hunk 11]
```diff
@@ -6,61 +6,64 @@
...
+          value="@confirmPassword?.AttemptedValue"
+          error-messages="@(confirmPassword?.Errors.FirstOrDefault()?.ErrorMessage)">
+          </v-text-field>
+        </form>
+      </v-col>
+    </v-row>
+  </v-card-text>
+  <v-card-actions>
+    @await RenderSectionAsync("AfterRegister", required: false)
+    <v-btn type="submit" form="form" x-large block color="success"><span>@T["Create account"]</span></button>
```
Comment (jptissot) [inline src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml:54]: v-btn

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml (hunk 5 above)
Comment (jptissot) [reply src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml:64]: It was a link before ;)  <a asp-route-area="OrchardCore.Users" asp-controller="Account" asp-action="Login">@T["Already have an account? Login"]</a>

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml (hunk 4 above)
Comment (jptissot) [reply src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml:56]: It should display the login link if the local login is not disabled or if there are some external providers

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml (hunk 9 above)
Comment (jptissot) [reply src/Themes/HackathonTheme/Views/OrchardCore.Users/Account/Login.cshtml:88]: I thought it was a link before ? `  <a asp-controller="Registration" asp-action="Register" asp-route-returnurl="@ViewData["ReturnUrl"]">@T["Register as a new user"]</a>`

File: src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml (hunk 11 above)
Comment (jptissot) [reply src/Themes/HackathonTheme/Views/OrchardCore.Users/Registration/Register.cshtml]: Can also remove the span

**Review by jptissot — APPROVED** [review]

**Pull request merged**